### ISSUE TITLE
Manticore produces a trace with a STOP when it should report a REVERT.

### ISSUE THREAD

**Issue by ggrieco-tob** [trailofbits/manticore#829]
### OS / Environment

Distributor ID:	ArchLinux
Description:	ArchLinux
Release:	rolling-release

### Manticore version

Last revision (2eb002593e894d97b5fc9f214d84d9b1d56e3d9a)

### Python version

Python 2.7.14

### Summary of the problem

Manticore produces a trace with a STOP when it should report a REVERT.

### Step to reproduce the behavior

This is a small solidity contract to reproduce this issue:

```solidity
contract BasicToken {

  function transfer(address _to, uint256 _value) public returns (bool) {
    require(_to != address(0));
    return true;
  }

}

contract C  {
    BasicToken public token;
    function C() public {
        token = new BasicToken();
    }

    function withdraw(address _to, uint _amount) external {
        token.transfer(_to, _amount);
    }

}
```

You can trigger this issue using the manticore cli:

```
$ manticore error.sol --contract C --txlimit 1                                                       
```

### Expected behavior

```
...

Transactions Nr. 1
Type: Call
From: 0xf522dfdc3f12cc0c75ffbff51e5876bf982e52b2 
To: 0x1bfa530d5d685155e98cd7d9dd23f7b6a801cfef 
Value: 0 (*)
Data: f3fef3a300000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000 (*)

Function call:
withdraw(0,0) -> REVERT 
```

### Actual behavior

```
...

Transactions Nr. 1
Type: Call
From: 0xf522dfdc3f12cc0c75ffbff51e5876bf982e52b2 
To: 0x1bfa530d5d685155e98cd7d9dd23f7b6a801cfef 
Value: 0 (*)
Data: f3fef3a300000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000 (*)

Function call:
withdraw(0,0) -> STOP (*)
```

**Comment by ggrieco-tob** [comment]
This script should reproduce this issue, according to @montyly:

```python

from manticore.ethereum import ManticoreEVM
from manticore.core.plugin import Plugin

m = ManticoreEVM()

class Tracer(Plugin):
    def did_evm_execute_instruction_callback(self, state, instruction, arguments, result):
        print "\t"*state.cpu.depth, hex(state.cpu.pc), instruction, state.cpu.stack


source_code = '''
contract REVERT{

    // should always return revert
    function test(){
        require(this.call(bytes4(keccak256('alwaysRevert()'))));
    }

    function alwaysRevert(){
        require(false);
    }
}
'''
user_account = m.create_account()
contract_account = m.solidity_create_contract(source_code, owner=user_account)

m.register_plugin(Tracer())
contract_account.test()

m.finalize()
```

**Comment by montyly** [comment]
My understanding of the bug is that REVERT should push '0' on the stack of the caller, but does not.

If you look at the execution of the code above, you will have (addresses may change according to your compiler version):

```
		0x87L REVERT [2679339091L, 100L]
	0x138L SWAP3 [4171824493L]
```
On the last instruction (0x138L SWAP3), the REVERT should have pushed `0x0` on the stack.

**Comment by ggrieco-tob** [comment]
Fixed in d1051f6.
